### PR TITLE
waitUntil, if given should return custom error message when condition returns always false

### PR DESCRIPTION
**Pull request by rashiq231**
## Proposed changes
waitUntil should return a custom error message if given, else should return a default error message when the condition always returns false ( not resolving to falsy)
[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)
This change is to resolve an issue reported in the #10495 , where waitUntil doesn't throw a custom error message if given.
With this change, we check for a condition if the given function always returns falsy and has a custom error message.
## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Review by christian-bromann — CHANGES_REQUESTED** [review]
I think the best solution is to do the following change in line 80 of `packages/webdriverio/src/utils/Timer.ts#L80`:

```patch
-                 return this._checkCondition(new Error('return value was never truthy'))
+                return this._checkCondition(new Error(TIMEOUT_ERROR))
```

Then the right timeout message should be thrown as defined in line 50

**Comment by rashiq231** [comment]
updated as per your comment. Could you please verify now?

File: packages/webdriverio/tests/commands/browser/waitUntil.test.ts [hunk 1]
```diff
@@ -191,6 +191,65 @@ describe('waitUntil', () => {
...
+        let val
+        // @ts-ignore uses expect-webdriverio
+        expect.assertions(2)
+        try {
+            val = await browser.waitUntil(()=>n, {
+                timeout: 500,
+                timeoutMsg: 'Custom error message',
+                interval: 200
+            }
+            )
```
Comment (christian-bromann) [inline packages/webdriverio/tests/commands/browser/waitUntil.test.ts:205]:
```suggestion
            val = await browser.waitUntil(() => n, {
                timeout: 500,
                timeoutMsg: 'Custom error message',
                interval: 200
            })
```

File: packages/webdriverio/tests/commands/browser/waitUntil.test.ts [hunk 2]
```diff
@@ -191,6 +191,65 @@ describe('waitUntil', () => {
...
+        let error
+        let val
+        // @ts-ignore uses expect-webdriverio
+        expect.assertions(2)
+        try {
+            val = await browser.waitUntil(()=>n, {
+                timeout: 500,
+                interval: 200
+            }
+            )
```
Comment (christian-bromann) [inline packages/webdriverio/tests/commands/browser/waitUntil.test.ts:224]:
```suggestion
            val = await browser.waitUntil(() => n, {
                timeout: 500,
                interval: 200
            })
```

File: packages/webdriverio/tests/commands/browser/waitUntil.test.ts [hunk 3]
```diff
@@ -191,6 +191,65 @@ describe('waitUntil', () => {
...
+        } catch (err: any) {
+            error = err
+        } finally {
+            expect(error).toBeUndefined()
+            expect(val).toBe(n)
+        }
+    })
+
```
Comment (christian-bromann) [inline packages/webdriverio/tests/commands/browser/waitUntil.test.ts:252]: We have this test already in line 168
```suggestion
```

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Comment by christian-bromann** [comment]
@rashiq231 I think we need to update one more unit test:

```
 FAIL  packages/webdriverio/tests/utils/Timer.test.ts > timer > promise > should be rejected without promise
AssertionError: expected Error: timeout to match object Error: return value was never truthy
  - Expected  - 1
  + Received  + 1

    Error {
  -   message: 'return value was never truthy',
  +   message: 'timeout',
    }
```

**Comment by rashiq231** [comment]
okay... I will update the test case for Timer.
But, a little clarification, shouldn't we need to maintain the existing error message - 'return value was never truthy', when ever the function returns false instead of resolving to false?
Shouldn't the timeout error which we updated in the timer should occur when the given time is exhausted?

**Comment by christian-bromann** [comment]
The docs implicitly say:

> It expects a condition and waits until that condition is fulfilled with a truthy value to be returned.

So there is no reason to have an error message stating the obvious and use the timeout error because that is what happens at the end: the timer times out because no truthy value was returned.

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Comment by christian-bromann** [comment]
Congratulations on your first WebdriverIO contribution! This project can't live without the participation of the community. We would love to see more from you, so let us know if we can help to find interesting areas for you to contribute to. We appreciate you 🙏 ❤️

**Comment by rashiq231** [comment]
@christian-bromann , Thank you for your help and support :)

**Pull request merged**